src/LDNS.xs

```diff
@@ -1375,12 +1375,15 @@ packet_edns_version(obj,...)
 # -------------------
 # Get/set EDNS data
 # 
-# Beware, this code can only take a unique U32 parameter which means it 
-# is not a full implementation of EDNS data but it is enough for our 
+# This function will set OPT RDATA in a packet when given a parameter,
+# otherwise it will return the entire OPT RDATA of the packet (if any).
+#
+# Beware, when setting OPT RDATA, this code can only take a unique U32 parameter
+# which means it is not a full implementation of EDNS data but it is enough for our
 # current purpose. It can only deal with option codes with OPTION-LENGTH=0
 # (see 6.1.2 section of RFC 6891) which means OPTION-DATA is always empty.
```

Comment:
In the description of this PR it says "This allows to check for all OPTION fields (OPTION-CODE, OPTION-LENGTH, OPTION-DATA) in an EDNS response packet", but the section above seems to say that OPTION-DATA is always empty. Or do I misread something?

Reply:
From what I understand:

* If used as a getter, this function will return the OPT pseudo-RR’s RDATA as a string of bytes (which indeed means that one can check each individual EDNS option, including data, but this needs to be done by the caller).
* If used as a setter, this function takes a second argument, a 32-bit integer. The network-order byte representation of that integer is taken as the byte string to be set as the new OPT pseudo-RR's RDATA. Assuming the integer passed as a second argument is a multiple of 65 536 (which is the case in the [only use of that function I've found in Zonemaster-Engine](https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Test/Nameserver.pm#L1175)), that effectively sets an empty EDNS option of the given code. For example, to set an empty option of code 137 (0x89), one should pass 137 × 65 536 = 8 978 432 (or 0x890000).

@tgreenx Feel free to replace this method’s current documentation-comment with my wording if you think it describes better what it does.

Reply:
@marc-vanderwal Yes, this is it mostly.

When this function is called without any argument, it will be used as a getter. Otherwise, if given one argument, it will be used as a setter. This paragraph specifies the limitation of the case when the function is called as a setter (i.e., when given an argument). I modified it just a bit to be even more explicit by adding "`when setting OPT RDATA`". Do you think it requires a change in the documentation?

Also, consider that this PR is only a quick fix for the issue at hand. In a future PR, I will propose to separate the setter/getter functionality in distinct functions (e.g., `edns_set_data` and `edns_get_data`), as well as proposing a full implementation of both (i.e., ability to detect several OPTIONS in one OPT pseudo-RR, ability to specify OPTION-LENGTH and OPTION-DATA in the setter, etc).

Reply:
I agree, this PR should be just a quick fix and a proper refactoring of that function is out of scope.

However it is idiomatic for object-oriented Perl code to have `$obj->some_property()` as a getter and `$obj->some_property($new_value)` as a setter, so there is nothing wrong with keeping that convention.

Reply:
I have rephrased the documentation. Please re-review.